core-bundle/src/Routing/ContentUrlGenerator.php

```diff
@@ -0,0 +1,191 @@
+            }
+
+            // The original content has changed, parameters are not valid anymore
+            if ($targetContent !== $content && $target !== $content) {
+                $parameters = [];
+            }
+
+            $compiledRoute = $route->compile();
+            $optionalParameters = [];
+
+            if ($targetContent) {
+                foreach ($this->urlResolvers as $resolver) {
+                    foreach ($resolver->getParametersForContent($targetContent, $target) as $k => $v) {
+                        if (isset($parameters[$k])) {
+                            continue;
+                        }
+
+                        $optionalParameters[$k] = $v;
+                    }
+                }
+
+                $optionalParameters = array_intersect_key($optionalParameters, array_flip($compiledRoute->getVariables()));
+            }
+
+            return $this->urlCache[$cacheKey] = $this->urlGenerator->generate(
+                PageRoute::PAGE_BASED_ROUTE_NAME,
+                [...$optionalParameters, ...$parameters, RouteObjectInterface::ROUTE_OBJECT => $route],
+                $referenceType,
+            );
+        } catch (ExceptionInterface $exception) {
+            $this->urlCache[$cacheKey] = $exception;
+
+            throw $exception;
+        }
+    }
+
+    public function setContext(RequestContext $context): void
+    {
+        $this->urlGenerator->setContext($context);
+    }
+
+    public function getContext(): RequestContext
+    {
+        return $this->urlGenerator->getContext();
+    }
+
+    public function reset(): void
+    {
+        $this->urlCache = [];
+    }
+
+    /**
+     * @throws ExceptionInterface
+     */
+    private function resolveContent(object ...$contents): array
+    {
+        foreach ($this->urlResolvers as $resolver) {
+            $result = $resolver->resolve($contents[0]);
+
+            if ($result->isAbstained()) {
+                continue;
+            }
+
+            if ($result->hasTargetUrl()) {
+                return [$result->getTargetUrl()];
+            }
+
+            if ($result->isRedirect()) {
+                return $this->resolveContent($result->content);
+            }
+
+            return $this->resolveContent($result->content, ...$contents);
+        }
+
+        if ($contents[0] instanceof PageModel) {
+            return $contents;
+        }
+
+        $this->throwRouteNotFoundException($contents[0]);
+    }
+
+    private function getRouteKey(object $content): string
+    {
+        if (is_subclass_of($content, Model::class)) {
+            return sprintf('%s.%s', $content::getTable(), $content->id);
+        }
+
+        try {
+            $metadata = $this->entityManager->getClassMetadata($content::class);
+        } catch (MappingException) {
+            $metadata = null;
+        }
+
+        if (null === $metadata) {
+            return $content::class.'->'.spl_object_hash($content);
```

Comment:
```suggestion
            return $content::class.'->'.spl_object_id($content);
```

Slightly faster because one C call can be saved :) https://github.com/php/php-src/blob/c76ec62518832a200654bc4fba5e3fcda978333d/ext/spl/php_spl.c#L651-L680

Reply:
fixed in 8fd38de8a0851c25e2567b731ddc9edc296d6725